guest/filter/filter.go

```diff
@@ -115,3 +115,57 @@ func (n *nodeInfo) lazyNode() proto.Node {
 	n.node = &internalproto.Node{Msg: &msg}
 	return n.node
 }
+
+type PodInfo struct {
+	pod proto.Pod
+}
+
+func (p *PodInfo) GetApiVersion() string {
+	return p.lazyPod().GetApiVersion()
+}
+
+func (p *PodInfo) GetKind() string {
+	return p.lazyPod().GetKind()
+}
+
+func (p *PodInfo) GetName() string {
+	return p.lazyPod().GetName()
+}
+
+func (p *PodInfo) GetNamespace() string {
+	return p.lazyPod().GetNamespace()
+}
+
+func (p *PodInfo) GetResourceVersion() string {
+	return p.lazyPod().GetNamespace()
+}
+
+func (p *PodInfo) GetUid() string {
+	return p.lazyPod().GetUid()
+}
+
+func (p *PodInfo) Pod() proto.Pod {
+	return p.lazyPod()
+}
+
+func (p *PodInfo) Spec() *protoapi.PodSpec {
+	return p.lazyPod().Spec()
+}
+
+func (p *PodInfo) Status() *protoapi.PodStatus {
+	return p.lazyPod().Status()
+}
+
+// lazyPod lazy initializes pod from imports.Pod.
+func (p *PodInfo) lazyPod() proto.Pod {
+	if pod := p.pod; pod != nil {
+		return pod
+	}
+
+	var msg protoapi.Pod
+	if err := imports.Pod(msg.UnmarshalVT); err != nil {
+		panic(err.Error())
+	}
+	p.pod = &internalproto.Pod{Msg: &msg}
```

Comment:
@sanposhiho 
I am not sure how to implement here. I would appreciate it if you could leave any comments.

Reply:
Ok, I will leave some comment:)

Reply:
We have to define another host function for k8sApiTargetPodFn in imports package. So, L166 is incorrect now.